### PR TITLE
Adds table view for excluded sites in Rewards Settings

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/3657
UI PR: https://github.com/brave/brave-ui/pull/443

A lot of code was removed related to getting the excluded publishers length number, now we just check the length of the excluded ContentSiteList we get back from the publisher_db.

Screen Shot 2019-04-10 at 4 26 54 PM

Screen Shot 2019-04-10 at 4 27 07 PM

Screen Shot 2019-04-10 at 4 27 17 PM

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. Start Brave, enable Rewards.
2. Add a few sites to the AC table.
3. Exclude a single site
4. Confirm that the "Show All" button is present in the lower right hand corner of the settings box.
5. Click on it and confirm that the three AC sites are in the supported sites table and that the single excluded site is in the excluded table.
6. Under Supported Sites in the modal, remove another site.
7. Confirm the excluded count increments and the site is now present in that table.
8. Confirm that under the excluded view that restoring a site/restore all removes the respective sites from the excluded table.
9. Confirm that any restored sites can be re-added to the AC table.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: browser/ui/webui/brave_rewards_ui.cc [hunk 1]
```diff
@@ -629,8 +631,19 @@ void RewardsDOMHandler::OnAutoContributePropsReady(
       props->reconcile_stamp,
       props->contribution_non_verified,
       props->contribution_min_visits,
+      false,
       base::Bind(&RewardsDOMHandler::OnContentSiteList,
                  weak_factory_.GetWeakPtr()));
+  rewards_service_->GetContentSiteList(
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_rewards_ui.cc]: this shouldn't happen here as now we are getting exclude list for other cases as well

Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: please create separate function for exclude

File: browser/ui/webui/brave_rewards_ui.cc [hunk 2]
```diff
@@ -732,6 +737,14 @@ void RewardsDOMHandler::RestorePublishers(const base::ListValue *args) {
   }
 }
 
+void RewardsDOMHandler::RestorePublisher(const base::ListValue *args) {
+  if (rewards_service_) {
+    std::string publisherKey;
+    args->GetString(0, &publisherKey);
+    rewards_service_->RestorePublisher(publisherKey);
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_rewards_ui.cc]: please add callback which will tell UI that publisher is restored and they need to fetch new lists

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 3]
```diff
@@ -513,6 +490,44 @@ bool PublisherInfoDatabase::InsertOrUpdateActivityInfos(
...
+
+  bool initialized = Init();
+  DCHECK(initialized);
+
+  if (!initialized) {
+    return false;
+  }
+
+  // We will use every attribute from publisher_info
+  std::string query = "SELECT * FROM publisher_info "
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/publisher_info_database.cc]: why do we need activity info? We just need to list all publishers in publisher_info that has exclude set to 1

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 4]
```diff
@@ -602,6 +617,7 @@ bool PublisherInfoDatabase::GetActivityList(
   }
 
   if (filter.excluded != ledger::EXCLUDE_FILTER::FILTER_ALL &&
+      filter.excluded != ledger::EXCLUDE_FILTER::FILTER_EXCLUDED &&
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/publisher_info_database.cc]: why is this needed?

Comment (ryanml) [reply components/brave_rewards/browser/publisher_info_database.cc]: leftover, removed

File: components/brave_rewards/browser/publisher_info_database.cc (hunk 3 above)
Comment (ryanml) [reply components/brave_rewards/browser/publisher_info_database.cc]: yep, we can simplify and just use `publisher_info`, done

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 5]
```diff
@@ -504,11 +504,7 @@ bool PublisherInfoDatabase::GetExcludedList(
   }
 
   // We will use every attribute from publisher_info
-  std::string query = "SELECT * FROM publisher_info "
-                      "WHERE publisher_info.publisher_id NOT IN "
-                      "(SELECT publisher_id FROM activity_info "
-                      "WHERE activity_info.publisher_id = publisher_info.publisher_id) "
-                      " AND publisher_info.excluded = 1";
+  std::string query = "SELECT * FROM publisher_info";
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/publisher_info_database.cc]: we still need ` WHERE excluded = 1`

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 6]
```diff
@@ -504,11 +504,7 @@ bool PublisherInfoDatabase::GetExcludedList(
   }
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/publisher_info_database.cc]: let's move this function `PublisherInfoDatabase::GetExcludedList` under `PUBLISHER INFO` section

File: components/brave_rewards/browser/publisher_info_database.cc (hunk 5 above)
Comment (ryanml) [reply components/brave_rewards/browser/publisher_info_database.cc]: woops, mistake, fixed!

File: browser/ui/webui/brave_rewards_ui.cc (hunk 1 above)
Comment (ryanml) [reply browser/ui/webui/brave_rewards_ui.cc]: done

File: components/brave_rewards/browser/publisher_info_database.cc (hunk 6 above)
Comment (ryanml) [reply components/brave_rewards/browser/publisher_info_database.cc]: done

File: browser/ui/webui/brave_rewards_ui.cc [hunk 7]
```diff
@@ -644,31 +651,46 @@ void RewardsDOMHandler::OnAutoContributePropsReady(
...
     brave_rewards::RewardsService* rewards_service,
     std::string publisher_id,
     bool excluded) {
-  if (rewards_service_)
-    rewards_service_->GetExcludedPublishersNumber(
-        base::Bind(&RewardsDOMHandler::OnGetExcludedPublishersNumber,
-                   weak_factory_.GetWeakPtr()));
+  if (rewards_service_) {
+    base::ListValue args;
+    GetExcludedSites(&args);
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_rewards_ui.cc]: this change should be propagated to JS and then JS should call `GetExcludedSites`

File: browser/ui/webui/brave_rewards_ui.cc [hunk 8]
```diff
@@ -746,6 +768,14 @@ void RewardsDOMHandler::RestorePublishers(const base::ListValue *args) {
   }
 }
 
+void RewardsDOMHandler::RestorePublisher(const base::ListValue *args) {
+  if (rewards_service_) {
+    std::string publisherKey;
+    args->GetString(0, &publisherKey);
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_rewards_ui.cc]: please use new approach for getting this value:
`CHECK_EQ(1U, args->GetSize());` (at the top of the function)
`args->GetList()[0].GetString()`

File: components/resources/brave_components_strings.grd [hunk 9]
```diff
@@ -517,6 +517,13 @@
       <message name="IDS_BRAVE_UI_PENDING_CONTRIBUTION_EMPTY" desc="Text for empty pending table">No pending contributions…</message>
       <message name="IDS_BRAVE_UI_PENDING_CONTRIBUTION_REMOVE_ALL" desc="Text for remove all button">Remove All</message>
 
+      <message name="IDS_BRAVE_UI_EXCLUDED_SITES_TEXT" desc="">Excluded Sites</message>
```
Comment (NejcZdovc) [inline components/resources/brave_components_strings.grd]: please add desc for all translations

File: components/brave_rewards/resources/ui/reducers/publishers_reducer.ts [hunk 10]
```diff
@@ -19,16 +19,12 @@ const publishersReducer: Reducer<Rewards.State | undefined> = (state: Rewards.St
...
-    case types.ON_EXCLUDED_PUBLISHERS_NUMBER: {
+    case types.ON_EXCLUDED_LIST: {
       state = { ...state }
-      let num = parseInt(action.payload.num, 10)
-
-      if (isNaN(num)) {
-        num = 0
+      if (action.payload.list) {
+        state.excludedList = action.payload.list
+        chrome.send('brave_rewards.getContributionList')
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/ui/reducers/publishers_reducer.ts]: why do we need to call contribution list here? If I am tracking it correctly `ON_EXCLUDED_LIST` is called when you call getter for excluded list

Comment (NejcZdovc) [reply components/brave_rewards/resources/ui/reducers/publishers_reducer.ts]: oh I see. It's like this because `OnExcludedSitesChanged` goes through the same flow. Exclude should have callback that would tell UI to fetch new excluded list and AC list. This way we don't get AC list unnecessary

File: browser/ui/webui/brave_rewards_ui.cc (hunk 7 above)
Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: done

File: browser/ui/webui/brave_rewards_ui.cc (hunk 8 above)
Comment (NejcZdovc) [reply browser/ui/webui/brave_rewards_ui.cc]: done

File: components/brave_rewards/resources/ui/reducers/publishers_reducer.ts (hunk 10 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/ui/reducers/publishers_reducer.ts]: done
